packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1705,20 +1860,20 @@ describe("BnsConnection", () => {
     // then claim, offer, claim - 2 closed, 1 open
     {
       const post = await claimSwap(connection, profile, faucet, id2, preimage1);
-      await post.blockInfo.waitFor(info => info.state === BcpTransactionState.InBlock);
+      await post.blockInfo.waitFor(info => info.state !== TransactionState.Pending);
     }
 
     // start to watch
     const liveView = asArray(connection.watchSwap(rcptQuery));
 
     const post3 = await openSwap(connection, profile, faucet, recipientAddr, preimage3);
-    const blockInfo3 = await post3.blockInfo.waitFor(info => info.state === BcpTransactionState.InBlock);
-    const id3 = (blockInfo3 as BcpBlockInfoInBlock).result! as SwapIdBytes;
+    const blockInfo3 = await post3.blockInfo.waitFor(info => info.state === TransactionState.Succeeded);
```

Comment:
Again, I don't like this, as it will hang forever if our call errors, with no useful diagnostics.
How about waiting for `Succeeded|Failure`, and then asserting it was a success, where we can easily dump out the tendermint logs for debugging.

There are dozens of such cases, I will stop commenting on them, but I do see as it is, if there are any DeliverTx failures when I do the weave v0.10 upgrade, I will end up with hanging tests and no error messages